pallets/automation-time/src/lib.rs

```diff
@@ -445,19 +445,19 @@ pub mod pallet {
 		/// * `DuplicateTask`: There can be no duplicate tasks.
 		/// * `TimeTooFarOut`: Execution time or frequency are past the max time horizon.
 		/// * `TimeSlotFull`: Time slot is full. No more tasks can be scheduled for this time.
-		#[pallet::weight(<T as Config>::WeightInfo::schedule_xcmp_task_full(execution_times.len().try_into().unwrap()))]
+		#[pallet::weight(<T as Config>::WeightInfo::schedule_xcmp_task_full(schedule.number_of_executions()))]
 		pub fn schedule_xcmp_task(
 			origin: OriginFor<T>,
 			provided_id: Vec<u8>,
-			execution_times: Vec<UnixTime>,
+			schedule: ScheduleParam,
```

Comment:
@chrisli30 API change number 1
`schedule_xcmp_task` extrinsic accepts a ScheduleParam enum instead of vec of execution times.

Reply:
yeah I’ve checked it out from the javascript tests here. https://github.com/OAK-Foundation/OAK-JS-SDK/pull/20#pullrequestreview-1136725349

This enum design is human readable and clearer. 👍